days/src/day4/Day4.kt

```diff
@@ -0,0 +1,62 @@
+package day4
+
+import Day
+import java.io.File
+
+class Day4(private val passports: List<Map<String, String>>) : Day {
+    enum class PassportField {
+        BYR, IYR, EYR, HGT, HCL, ECL, PID;//, CID
+
+        override fun toString() = super.toString().toLowerCase()
+        private val _eyeColors = listOf("amb", "blu", "brn", "gry", "grn", "hzl", "oth")
+        fun validate(value: String) = when (this) {
+            BYR -> value.toInt() in 1920..2002
```

Comment:
If they extend this problem you'll probably wan to functionalize the validations just to make it easier to read, edit, and test

Reply:
Easy to do